### DISCUSSION TITLE
User options to CVC5 solver

### DISCUSSION THREAD

**Discussion by Avgor46** [sosy-lab/java-smt#389] [Q&A] [answered]
Is there a way to pass user parameters (e.g. "solve-bv-as-int") to the СVС5 solver? According to the source code this is not possible due to fact that `CVC5SolverContext.setSolverOptions` method is private, am I right? Maybe it would be nice to add such feature.

**Comment by kfriedberger** [comment, marked as answer]
No, there was no way to provide options fro CVC5 directly.
We will add this feature: https://github.com/sosy-lab/java-smt/pull/390

**Comment by baierd** [comment]
You can now define solver specific options for CVC5 using the `furtherOptions` configuration option on the main branch.
It can be used by defining a configuration and adding `furtherOptions` as seen in this example for the CVC5 option `tlimit-per`: `yourConfig.setOption("solver.cvc5.furtherOptions", "tlimit-per=1000")....)` 

(`furtherOptions` description: Further options that will be passed to CVC5 in addition to the default options. Format is 'key1=value1,key2=value2')

**Reply by kfriedberger** [reply]
We will release a new version of JavaSMT within the next days.

**Reply by kfriedberger** [reply]
[JavaSMT 5.0.1](https://github.com/sosy-lab/java-smt/releases/tag/5.0.1) is released with this feature.
